packages/sitecore-jss-nextjs/src/middleware/editing-render-middleware.ts

```diff
@@ -0,0 +1,160 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+import { AxiosDataFetcher } from '@sitecore-jss/sitecore-jss';
+import { EditingData } from '../sharedTypes/editing-data';
+import { EditingDataService, editingDataService } from '../services/editing-data-service';
+import { QUERY_PARAM_SECURITY_TOKEN } from '../services/editing-data-service';
+import { getPublicUrl, getSitecoreSecurityToken } from '../utils';
+
+export interface EditingRenderMiddlewareConfig {
+  /**
+   * The `AxiosDataFetcher` instance to use for API requests.
+   * @default new AxiosDataFetcher()
+   * @see AxiosDataFetcher
+   */
+  dataFetcher?: AxiosDataFetcher;
+  /**
+   * The `EditingDataService` instance to use.
+   * This would typically only be necessary if you've got a custom `EditingDataService` instance (e.g. using a custom API route).
+   * By default, this is `editingDataService` (an `EditingDataService` singleton).
+   * @default editingDataService
+   * @see EditingDataService
+   */
+  editingDataService?: EditingDataService;
+  /**
+   * Function used to determine route/page URL to render.
+   * This may be necessary for certain custom Next.js routing configurations.
+   * @param {string} serverUrl The root server URL e.g. 'http://localhost:3000'
+   * @param {string} itemPath The Sitecore relative item path e.g. '/styleguide'
+   * @returns {string} The URL to render
+   * @default `${serverUrl}${itemPath}`
+   */
+  renderRouteResolver?: (serverUrl: string, itemPath: string) => string;
+}
+
+/**
+ * Middleware / handler for use in the editing render Next.js API route (e.g. '/api/editing/render')
+ * which is required for Sitecore Experience Editor support.
+ */
+export class EditingRenderMiddleware {
+  private editingDataService: EditingDataService;
+  private dataFetcher: AxiosDataFetcher;
+  private renderRouteResolver: (serverUrl: string, itemPath: string) => string;
```

Comment:
For me this name means that this function responsible for rendering of route, 
May be better to rename it to: `resolveRouteUrl`?

Reply:
@sc-illiakovalenko Agree, better name for it. Changed.